pages/genre/_itemId/index.vue

```diff
@@ -47,45 +47,43 @@
 
 <script lang="ts">
 import Vue from 'vue';
-import { mapActions } from 'vuex';
+import { mapActions, mapGetters } from 'vuex';
 import { BaseItemDto } from '@jellyfin/client-axios';
 import itemHelper from '~/mixins/itemHelper';
 
 export default Vue.extend({
   mixins: [itemHelper],
-  async asyncData({ params, $api, $auth }) {
-    const genre = (
-      await $api.userLibrary.getItem({
-        userId: $auth.user?.Id,
-        itemId: params.itemId
-      })
-    ).data;
-
-    return { genre };
+  async asyncData({ params, $libraries }) {
+    await $libraries.fetchItem(params.itemId);
   },
   data() {
     return {
-      genre: [] as BaseItemDto,
-      items: [] as BaseItemDto[]
+      itemIds: [] as string[]
     };
   },
   async fetch() {
-    this.items = (
-      await this.$api.items.getItems({
-        userId: this.$auth.user?.Id,
-        genreIds: [this.$route.params.itemId],
-        includeItemTypes: [this.$route.query.type.toString()],
-        recursive: true,
-        sortBy: 'SortName',
-        sortOrder: 'Ascending'
-      })
-    ).data.Items;
+    this.itemIds = await this.$libraries.fetchItems({
```

Comment:
This would move to a store too.

A basic `record<string, string[]>` in a `genres` store or something.

Reply:
key being an item like movie and the value array being the genres ids?

Reply:
Something like that, but that would be for another PR down the line. No need to over-extend this one, it's already touching a lot of files and changing a fair number of things, I'd rather avoid doing this in one PR (If only for being able to split up the work among ourselves. You doing all of this in one PR instead of doing one PR per page is already blocking us from making changes or improvements to these ourselves :smile:  )